124975981 Correction of Absorptivity Assignment in Movable Insulation

RKStrand opened the pull request:
## Pull request overview

This defect came about be a discovery that the movable insulation code was not making any distinction between the solar and thermal absorptivity for OUTSIDE movable insulation.  The code was basically always using the solar absorptivity when in one situation it should have used the thermal absorptivity.  This problem was fixed.  In addition, during the testing of this fix, it was discovered that the material properties it was using for solar absorptivity were only correct if the material layer was a glazing layer.  For other material layers, the calculation of solar absorptivity was incorrect.  This was fixed for both inside and outside movable insulation.  It should be noted that even though this was established for inside movable insulation that this still does not allow the proper modeling of transparent movable insulation on the INSIDE.  This would be a future enhancement idea though not very high priority as there is not much interest in movable insulation and generally transparent insulation only makes sense on the exterior of the building.  This addresses GitHub Issue #5757 as originally stated and the additional issues as well.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ x ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ x ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This work started with the correction of GitHub Issue #5757 and then corrected additional issues that were discovered along the way.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] Open windows IDF Editor with modified IDD to check for errors
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

RKStrand committed:
Bug fix code

Fix includes problem with not getting the thermal absorptivity when it
was needed and not solar absorptivity; also includes another related
fix where certain material types (glass layers) have certain properties
and others do not; this was a result of the transparent insulation
installation and reorganization of data in EnergyPlus (this is now
fixed)

RKStrand committed:
Additional fix and docs mod

Ported a change made to the outside movable insulation calculation for
solar absorptivity to the inside so that the algorithms match properly
and also added a statement to the docs so that users know how solar
absorptivity is being calculated.

RKStrand committed:
Addition of unit tests

Two unit tests were added to verify that the code work in the movable
insulation is pulling the correct material properties.  For non-glazing
materials, it uses simply the solar absorptivity.  For glazing
materials, it subtracts the transmissivity and the front side solar
reflectance from unity to calculate the solar absorptivity.

RKStrand commented:
@mjwitte @Myoldmopar Could one of you also verify this fix?  I made the mistake of changing the title.  So, this was originally GitHub Issue #5757 and I should have left it as that.  When I made the pull request, I made the mistake of changing the title and this now is being called #5762.  My apologies for that confusion.  Other than that, this will hopefully be another relatively straightforward fix.  It includes a couple of interrelated issues, two unit tests, and a documentation clarification.

Myoldmopar commented on doc/input-output-reference/src/overview/group-advanced-surface-concepts.tex:
```diff
@@ -1,4 +1,10 @@
-\section{Group -- Advanced Surface Concepts}\label{group-advanced-surface-concepts}
+\documentclass[10pt]{article}
```
@RKStrand This standalone LaTeX preamble will have to be removed back out so that the documentation can be built as part of the full pdf.

RKStrand replied:
@Myoldmopar No idea how this got in there.  I simply edited the text on line 232/238 and nothing else.  I will eliminate these lines and the one at the end and hope they don't magically pop up again.

Myoldmopar commented on doc/input-output-reference/src/overview/group-advanced-surface-concepts.tex:
```diff
@@ -1877,3 +1883,5 @@ \subsubsection{Inputs}\label{inputs-17}
     Lshaped Zone:South Wall,Lshaped Zone:East Wall,0.101310,
   <snip>
 \end{lstlisting}
+\]
```
Similar to [above](https://github.com/NREL/EnergyPlus/pull/5762/files#r71266857)...

Myoldmopar commented on src/EnergyPlus/HeatBalanceMovableInsulation.cc:
```diff
@@ -177,8 +179,12 @@ namespace HeatBalanceMovableInsulation {
 			RoughIndexMovInsul = Material( Surface( SurfNum ).MaterialMovInsulExt ).Roughness;
-			AbsExt = max( 0.0, 1.0 - Material( Surface( SurfNum ).MaterialMovInsulExt ).Trans - Material( Surface( SurfNum ).MaterialMovInsulExt ).ReflectSolBeamFront );
-
+			MaterialGroupNum = Material( Surface( SurfNum ).MaterialMovInsulExt ).Group;
```
Minor comments:
- Consider defining `MaterialGroupNum` directly where it is used, no need to define variables at the top of each function like we had to do with FORTRAN.  Making that a habit can ensure the variable is scoped properly, which can aid in the compiler auto-optimizing the code.
- Consider cleaning up empty and unnecessary comment block sections from the old FORTRAN template (`DERIVED TYPE DEFINITIONS`, etc.) as you go, to slowly clean the code up.
- There are a number of lookups for the `Surface( SurfNum ).MaterialMovInsulExt`.  The compiler _may_ be optimizing these away, but it may be better to reduce these calls down by using a single reference.  Add in `auto & thisSurface( Surface( SurfNum ) )`, and then change each of those `Surface( SurfNum )` calls to just be `thisSurface`. Like I said, it may not affect the compiler, but it will make a lighter looking block of code.

Myoldmopar commented on src/EnergyPlus/HeatBalanceMovableInsulation.cc:
```diff
@@ -140,7 +140,8 @@ namespace HeatBalanceMovableInsulation {
 
 		// USE STATEMENTS:
-		// na
+		using DataHeatBalance::WindowGlass;
```
In general, I'd say just go ahead and write `DataHeatBalance::WindowGlass` in the below code instead of a using statement.  For this small of a function, it's not a big deal either way, but with larger functions, if there are many using statements, it gets hard to understand where each variable is coming from.

RKStrand replied:
@Myoldmopar I'll take care of these other comments as soon as I can.  Not sure how a merge conflict got introduced, but glad you know what it is and how to resolve it.

Myoldmopar commented on src/EnergyPlus/HeatBalanceMovableInsulation.cc:
```diff
@@ -245,8 +253,15 @@ namespace HeatBalanceMovableInsulation {
 			HMovInsul = 1.0 / ( MovInsulSchedVal * Material( Surface( SurfNum ).MaterialMovInsulInt ).Resistance );
-			AbsInt = Material( Surface( SurfNum ).MaterialMovInsulInt ).AbsorpSolar;
 
+			MaterialGroupNum = Material( Surface( SurfNum ).MaterialMovInsulInt ).Group;
```
Similar comments to above.

Myoldmopar commented:
This looks good.  The unit tests are nice and to the point.  I made a few pretty minor comments.  If you could make those changes, or at least respond to my thoughts, and then pull in develop (resolving the merge conflict along the way), this should be able to go in.

RKStrand committed:
Fixes to address review comments

Fixes to issues raised by @Myoldmopar during the review process.  This
commit should address all of the issues mentioned.

RKStrand committed:
Merge branch 'develop' into 124975981-Outside-Movable-Insulation-Absorptivity-Issue

# Conflicts:
#	tst/EnergyPlus/unit/CMakeLists.txt

RKStrand commented:
@Myoldmopar Ok, I think I have addressed all of the issues you brought up in the latest commit that I made.  And I made another commit which I think addresses the merge conflict?  I'm not sure why it is considered a conflict unless I missed something along the way.  Let me know if I missed anything or if I messed anything up.  Oh, and by the way, I just happened to fix another bug along the way that I missed earlier.

Myoldmopar commented on tst/EnergyPlus/unit/CMakeLists.txt:
```diff
@@ -51,7 +51,11 @@ set( test_src
   General.unit.cc
   GroundHeatExchangers.unit.cc
   HeatBalanceManager.unit.cc
+<<<<<<< HEAD
```
This is the problem @RKStrand; some git merge conflict markers were left in the source after the merge.....  Looks like the resolution is to simply **keep** both unit.cc files listed here and delete the markers.  I'll do that now actually...

Myoldmopar committed:
Resolve conflict, remove markers.

Myoldmopar commented:
OK, this should now build fine.  You mention that this addresses multiple bugs.  Are there multiple _Github tickets_ to go along with those bugs, or just the one Github issue?

RKStrand commented:
@Myoldmopar This was a single GitHub issue.  I happened to see another obvious one line issue and I simply corrected it along with the other fixes for the bug that caused this GitHub issue in the first place rather than creating another GitHub issue.  Hope that was ok.
